mlir/include/Quantum/IR/QuantumOps.td

```diff
@@ -781,12 +781,15 @@ def AdjointOp : Op<QuantumDialect, "adjoint", [SingleBlock]> {
     let hasVerifier = 1;
 }
 
-def YieldOp : Op<QuantumDialect, "yield", [Pure, ReturnLike, Terminator]> {
+def YieldOp : Op<QuantumDialect, "yield",
+                [Pure, ReturnLike, Terminator, ParentOneOf<["AdjointOp"]>]> {
     let summary = "Return results from quantum program regions";
 
     let arguments = (ins Variadic<AnyTypeOf<[QuregType]>>:$results);
 
     let assemblyFormat = [{  attr-dict ($results^ `:` type($results))? }];
+
+    let builders = [OpBuilder<(ins), [{ /* nothing to do */ }]>];
```

Comment:
Ah is this builder needed for the trait then?

Reply:
Yes, it is. It results in a no-argument builder that is used somewhere in the trait.